### PR TITLE
tags: parse "line:" and use it for search

### LINKED ISSUE #1057: tags: take into account optional "line" field
When there are multiple entries in a tags file with the same pattern,
Vim will always go to the first one (`Bar.provision` in the example below).

```
provision       foo.py       /^    def provision(self, **kwargs):$/;"        m       line:1153       language:Python class:Foo
provision       foo.py       /^    def provision(self, **kwargs):$/;"        m       line:365        language:Python class:Bar
provision       foo.py       /^    def provision(self, request=None, **kwargs):$/;"  m       line:3564       language:Python class:Baz
```

It would be nice if Vim would take into account the optional `line:` field, to start the excmd from there.

An alternative might be to generate excmds that include the line number already, e.g. `1152;/^    def…/` for the first entry.  But from looking at the source in `tag.c` there seem to be places, where Vim looks for the excmd to start with a '/'?! (but that might be after splitting them)

### PR DESCRIPTION
**Pull request by blueyed**
Fixes https://github.com/vim/vim/issues/1057.

### CODE REVIEW

File: src/tag.c [hunk 1]
```diff
@@ -32,6 +32,8 @@ typedef struct tag_pointers
 #endif
     char_u	*tagkind;	/* "kind:" value */
     char_u	*tagkind_end;	/* end of tagkind */
+    char_u	*tagline;	/* "line:" value */
+    char_u	*tagline_end;	/* end of tagline */
```
Comment (blueyed) [inline src/tag.c:36]: `tagline_end` is not really used, but just `atoi`.
Maybe `tagline_end` could be used.

File: src/tag.c [hunk 2]
```diff
@@ -3332,7 +3348,15 @@ jumpto_tag(
 #endif
 #endif
 	    save_lnum = curwin->w_cursor.lnum;
-	    curwin->w_cursor.lnum = 0;	/* start search before first line */
+	    if (tagp.tagline)
+	    {
+		int tagline_num = atoi((char *)tagp.tagline);
+		curwin->w_cursor.lnum = tagline_num - 1;	/* start search before line */
```
Comment (blueyed) [inline src/tag.c:3354]: Needs error handling probably.
lnum should not become -1 etc.

**Comment by chrisbra** [comment]
so could you fix those TODOS and possibly add a test?

**Comment by brammool** [comment]
Please add a test.

**Comment by blueyed** [comment]
1. Should it go into src/testdir/test_tagjump.vim?
2. Any feedback on my comments/questions in the PR?

**Comment by chrisbra** [comment]
> Any feedback on my comments/questions in the PR?

I did not see any, besides your comments to fix some points...

**Comment by brammool** [comment]
Any update?

**Pull request closed without merging**